project/MAINTAINERS.md

```diff
@@ -83,6 +83,125 @@ this (see below "Who decides what?")
+
+A pull request can be in 1 of 5 distinct states, for each of which there is a
+corresponding label that needs to be applied. At any step of the way, a PR can
+be closed or return to an earlier state if needed (e.g. back to design review
+during code review).
+
+### States
+
+#### Triage (*no label*)
+
+Maintainers are expected to triage new incoming pull requests by adding
+the correct labels (e.g. `1-design-review`) potentially skipping some steps
+depending on the kind of pull request. Use common sense for judging.
+
+Checking for DCO should be done at this stage.
+
+PRs with new user-facing features should follow all steps detailed below.
+However, performance optimizations or internal features (e.g. changes in
+/project) can skip Design review and/or Documentation review steps.
+
+Bugfixes can skip Design review if the bug is trivial. However, design review
+on bugfixes does not always need to be skipped: if a behavior is being
+mistaken for a bug. Rule of thumb: if you are not sure apply `1-design-review`
+label. If you reproduced the bug yourself, apply `2-code-review` directly.
+
+Proposals with only documentation changes, should get the `1-design-review`
+applied.
+
+Other documentation-only PRs should get the `3-docs-review` label applied
+directly.
+
+If an owner, responsible for closing or merging, can be assigned to the PR,
+the better.
+
+#### Design review (`1-design-review`)
+
+Maintainers are expected to comment on the design of the pull request.
+Review of documentation is expected only in the context of design validation,
+not for stylistic changes.
+
+Ideally, documentation should reflect the expected behavior of the code.
+No code review should take place in this step.
+
+Once design is approved, a maintainer should remove this label and add
+the next appropriate label.
+
+In the case of Proposals with only documentation changes, the `3-docs-review`
+label should be applied.
+
+For changes that involve code, the `2-code-review` label should be applied.
+
+#### Code review (`2-code-review`)
+
+Maintainers are expected to review the code and ensure that it is good
+quality and in accordance with the documentation.
+
+If documentation is absent but expected, maintainers should ask for documentation.
+
+All tests should pass.
+
+Once code is approved (i.e. PR has a certain required number of LGTMs), a
+maintainer should remove this label and add the `3-docs-review` label in the
+case the PR contains changes to the documentation.
+
+For bugfixes or internal changes, the `4-merge` label should be applied.
+
+#### Documentation review (`3-docs-review`)
+
+Maintainers are expected to review the documentation in its bigger context,
+ensuring consistency throughout the entire documentation. They should ask
+for any editorial change that makes the documentation more consistent and
+easier to understand.
+
+Once documentation is approved, a maintainer should remove this label and
+add the `4-merge` label.
```

Comment:
This seems insufficient for most documentation reviews - where we also check that the change is factual. If the process for non-proposal doc-only PRs is to go straight to this state, then this state needs to include review to ensure what is written matches the code and design.